servers/zms/src/main/java/com/yahoo/athenz/zms/ZMSImpl.java

```diff
@@ -2607,6 +2607,53 @@ public Template getTemplate(ResourceContext ctx, String templateName) {
         return template;
     }
 
+    @Override
+    public DomainTemplateDetailsList getDomainTemplateDetailsList(ResourceContext ctx, String domainName) {
+        DomainTemplateDetailsList domainTemplateDetailsList = null;
+        List<TemplateMetaData> templateMetaDataList;
+        final String caller = "getDomainTemplateDetailsList";
+        metric.increment(ZMSConsts.HTTP_GET);
+        metric.increment(ZMSConsts.HTTP_REQUEST);
+        metric.increment(caller);
+        logPrincipal(ctx);
+        validateRequest(ctx.request(), caller);
+        validate(domainName, TYPE_DOMAIN_NAME, caller);
+
+        // for consistent handling of all requests, we're going to convert
+        // all domain into lower case
+
+        domainName = domainName.toLowerCase();
+        final String principalDomain = getPrincipalDomain(ctx);
+        Object timerMetric = metric.startTiming("getdomainrolemembers_timing", domainName, principalDomain);
+
+        List<TemplateMetaData> templateDomainMapping = dbService.getDomainTemplates(domainName);
+        if (templateDomainMapping != null) {
+            domainTemplateDetailsList = new DomainTemplateDetailsList();
+            templateMetaDataList = new ArrayList<>();
```

Comment:
there is no need to create this list again. we already have a list returned from dbservice. we should just add attributes and set that and return as part of our data. This way we're not just creating another set of objects which contains the same data as the original list. So just go through the list that we got on line 2629, update the objects and then on line 2650 we use that to set the meta data.

if we come across an entry that has not value then we should send it back since that shows something not right in the db - where we have a template that no longer exists in the server.

Reply:
good catch. Done. reused the same object that comes back from DBservice.